plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchPageSourceProvider.java

```diff
@@ -28,18 +28,21 @@
 import java.util.List;
 
 import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.plugin.elasticsearch.ElasticsearchTableHandle.Type.AGG;
 import static io.trino.plugin.elasticsearch.ElasticsearchTableHandle.Type.QUERY;
 import static java.util.Objects.requireNonNull;
 
 public class ElasticsearchPageSourceProvider
         implements ConnectorPageSourceProvider
 {
     private final ElasticsearchClient client;
+    private final ElasticsearchConfig config;
 
     @Inject
-    public ElasticsearchPageSourceProvider(ElasticsearchClient client)
+    public ElasticsearchPageSourceProvider(ElasticsearchClient client, ElasticsearchConfig config)
     {
         this.client = requireNonNull(client, "client is null");
+        this.config = requireNonNull(config, "config is null");
```

Comment:
Let's not store a reference to the config (since it's a mutable object).  Just extract the piece of informatin you need from it.